src/lib/auth.ts

```diff
@@ -57,58 +63,75 @@ export const ROLES: {
 };
 
 export const DEVICE_API_KEY_PERMISSIONS = [
-	`resin.device_type.get?describes_device/any(d:d/${matchesActor})`,
-	`resin.device.get?${matchesActor}`,
+	`resin.device_type.read?describes_device/canAccess()`,
+	`resin.device.read?${matchesActor}`,
 	`resin.device.update?${matchesActor}`,
-	`resin.supervisor_release.get?should_manage__device/any(d:d/${matchesActor})`,
-	`resin.application.read?${ownsDevice} or depends_on__application/any(da:da/${ownsDevice})`,
-	`resin.application_tag.read?application/any(a:a/${ownsDevice} or a/depends_on__application/any(da:da/${ownsDevice}))`,
-	`resin.device_config_variable.get?device/any(d:d/${matchesActor} or d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
+	`resin.supervisor_release.read?should_manage__device/canAccess()`,
+	`resin.application.read?owns__device/canAccess() or depends_on__application/canAccess()`,
+	`resin.application_tag.read?application/canAccess()`,
+	`resin.device_config_variable.read?device/canAccess()`,
 	`resin.device_config_variable.set?device/any(d:d/${matchesActor})`,
-	`resin.device_tag.get?device/any(d:d/${matchesActor})`,
+	`resin.device_tag.read?device/canAccess()`,
 	`resin.device_tag.set?device/any(d:d/${matchesActor})`,
-	`resin.application_config_variable.get?${applicationControlsDevice}`,
-	`resin.release.read?should_be_running_on__device/any(d:d/${matchesActor}) or ${belongsToApplicationOrIsManagedByDevice}`,
-	`resin.release_tag.read?release/any(r:r/should_be_running_on__device/any(d:d/${matchesActor}) or r/${belongsToApplicationOrIsManagedByDevice})`,
-	`resin.device_environment_variable.get?device/any(d:d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
-	`resin.device_environment_variable.all?device/any(d:d/${matchesActor})`,
-	`resin.application_environment_variable.get?${applicationControlsDevice}`,
+	`resin.application_config_variable.read?application/canAccess()`,
+	`resin.release.read?should_be_running_on__device/canAccess() or ${belongsToApplicationOrIsManagedByDevice}`,
```

Comment:
`${belongsToApplicationOrIsManagedByDevice}` could become `belongs_to__application/canAccess()` I think?

Reply:
Done, good spot - I was checking for reads that didn't already use `canAccess` which missed this one that did.. but only for one half